internal/expr/parser.go

```diff
@@ -677,6 +764,13 @@ func (p *Parser) parseOutputExpr() (*outputExpr, bool, error) {
 				if !parenCols && parenTypes {
 					return nil, false, errorAt(fmt.Errorf(`unexpected parentheses around types after "AS"`), p.lineNum, parenCol, p.input)
 				}
+				if starCountTypes(targetTypes) > 0 {
```

Comment:
Yep, it is partially because it is doing error handling and validation. We could move it out by I think it will have the same messy-ness / complexity.